app/src/main/java/edu/uc/hickmadc/customnotifications/service/GroupService.kt

```diff
@@ -0,0 +1,27 @@
+package edu.uc.hickmadc.customnotifications.service
+
+import edu.uc.hickmadc.customnotifications.dto.Group
+
+/**
+ * GroupService class. Handles adding groups to the groups tab.
+ *
+ * The GroupService class handles adding groups to the groups tabs via a connection to the database.
+ * However, this class currently holds some mocked data that emulates what would be stored or
+ * fetched from the database.
+ */
+class GroupService {
+
+    /**
+     * getAllGroups function. Essentially a fetchAll() function that returns all the existing
+     * groups.
+     * @return an arraylist of Group DTO objects.
+     */
+    fun getAllGroups() : ArrayList<Group>{
+        var groups : ArrayList<Group> = ArrayList()
+        //placeholder, should be replaced by database call/live data
+        groups.add(Group(name = "Group1", desc = "This data is", active = true, same_schedule = false, group_id = 1))
+        groups.add(Group(name = "Group2", desc = "set in the", active = false, same_schedule = false, group_id = 1))
+        groups.add(Group(name = "Group3", desc = "main view model", active = true, same_schedule = false, group_id = 2))
+        return groups
+    }
+}
```

Comment:
Suggestion 2: Created a Service layer and a corresponding kotlin class, and moved a fetchAll function down from the MainViewModel. This should keep the service layer separate from the MainViewModel, and also make it a bit easier to add database functionality down the road.
Source: https://docs.sonarqube.org/display/SONARQUBE45/Bad+Distribution+of+Complexity

Reply:
Good.